synapse/config/tls.py

```diff
@@ -70,6 +71,48 @@ def read_config(self, config):
 
         self.tls_fingerprints = list(self._original_tls_fingerprints)
 
+        # Whether to verify certificates on outbound federation traffic
+        self.federation_verify_certificates = config.get(
+            "federation_verify_certificates", False,
+        )
+
+        # Whitelist of domains to not verify certificates for
+        self.federation_certificate_verification_whitelist = {}
+        federation_certificate_verification_whitelist = config.get(
+            "federation_certificate_verification_whitelist", [],
+        )
+
+        # Store whitelisted domains in a hash for fast lookup
+        for domain in federation_certificate_verification_whitelist:
+            self.federation_certificate_verification_whitelist[domain] = True
+
+        # List of custom certificate authorities for federation traffic validation
+        self.federation_custom_ca_list = config.get(
+            "federation_custom_ca_list", None,
+        )
+
+        # Read in and parse custom CA certificates
+        if self.federation_custom_ca_list is not None:
+            certs = []
+            for ca_file in self.federation_custom_ca_list:
+                logger.debug("Reading custom CA certificate file: %s", ca_file)
+                try:
```

Comment:
you probably want `content = self.read_file(ca_file)` here, and you can skip some of the tedious exception handling.